edit/base.js

```diff
@@ -48,6 +63,19 @@ const editor = Object.assign(EventEmitter(), {
       customName || name || t('styleMissingName')
     } - Stylus`; // the suffix enables external utilities to process our windows e.g. pin on top
   },
+};
+
+editor.on('styleUpdated', (newStyle, reason) => {
+  if (reason === 'config') {
+    delete newStyle.sourceCode;
+    delete newStyle.sections;
+    delete newStyle.name;
+    delete newStyle.enabled;
+    Object.assign(editor.style, newStyle);
+    editor.updateLivePreview();
+  } else if (reason !== 'new') {
```

Comment:
Suppose L74 works fine, we should probably use a cleaner flow. The current event flow looks like this:
```
editor.save  -> styleChange(new) -> settings
             -> livePreview
styleUpdated -> styleChange(any)     -> settings
                styleChange(config)  -> livePreview
                styleChange(non-new) -> replaceStyle
```
It might be better to change it to
```
editor.save      -> livePreview
editor.save(new) -> settings
styleUpdated(any) -> settings
                  -> replaceStyle
styleUpdated(config) -> livePreview
```
i.e. we don't need this listener.

Reply:
I didn't try to change the flow, just couldn't resist removing the extra returns in such a short function. I look at the flow in your comment and I can't understand the difference just by looking so I'd say this is something that you can do in a separate PR rather than explain it to me. I'll get it when I look at the code, hopefully :-)